python/src/pyarrow/adapters/pandas.cc

```diff
@@ -1648,7 +1649,17 @@ inline Status ArrowSerializer<TYPE>::ConvertTypedLists(
         RETURN_NOT_OK(value_builder->Append(data, size));
       }
     } else if (PyList_Check(objects[i])) {
-      return Status::TypeError("Python lists are not yet supported");
+      PyAcquireGIL lock;
+      int64_t size;
+      std::shared_ptr<arrow::DataType> type;
+      RETURN_NOT_OK(list_builder.Append(true));
+      RETURN_NOT_OK(InferArrowType(objects[i], &size, &type));
+      if (type->type != field->type->type) {
```

Comment:
We may allow this in future but currently we are missing a matching converter or alternatively, a convenient cast function for other integer types.